Sources/Danger/File.swift

```diff
@@ -19,13 +19,8 @@ extension File {
 
 // MARK: - FileType
 
-public enum FileType: String, Equatable {
+public enum FileType: String, Equatable, CaseIterable {
     case h, json, m, markdown = "md", mm, pbxproj, plist, storyboard, swift, xcscheme, yaml, yml
-
-    @available(swift, deprecated: 4.2, message: "Replace with CaseIterable conformance")
-    static internal var allCases: [FileType] {
-        return [.h, .json, .m, .markdown, .mm, .pbxproj, .plist, .storyboard, .swift, .xcscheme, yaml, yml]
-    }
 }
```

Comment:
Any chance you can make this a backwards compatible change?

Reply:
@yhkaplan You should be able to wrap the property in `#if !(swift(>=4.2))`. 

Btw I noticed the `yaml` and `yml` cases in the array aren't prefixed with a dot... isn't that odd... or am I missing something?

Reply:
> yaml and yml cases in the array aren't prefixed with a dot

Really odd that the compiler didn't catch this... Will fix!

Reply:
I've added backwards compatibility (and added the missing periods) in the commit below. The only thing I'm wavering about still is what to set the CI to. (Also it looks like I forgot to set the .swift-version file)

I could set the macOS and Linux versions to Swift 4.2 and 4.1 respectively, but what do you think? Or we could wait to merge this until Swift 4.2 is available on Swiftenv, which may be soon.

Reply:
It looks like Swift 4.2 support was added to Swiftenv in the master branch on Sep 19, just needs to be released

https://github.com/kylef/swiftenv/commit/a949ff6ab8d1386f1bc99219da1ba860ea7de35a

Reply:
The install script for Swiftenv picks up the latest version which contains the Swift 4.2 installation link, so the failure point on Linux and macOS just seems to be Danger-JS right now.

```
Error:  TypeError: Cannot read property 'concat' of undefined
    at Object.exports.mergeResults
```